### PR TITLE
UHM-4998, migrate PCR tests

### PR DESCRIPTION
**Pull request by cioan**
Hi @mogoodrich , here is the PR for migrating the PCR tests. Thanks!

### CODE REVIEW

File: etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                    uuid)
+            select 
+                    i.person_id, 
+                    r.encounter_id, 
+                    r.source_encounter_date as obsDatetime, 
+                    1 as locationId, 
+                    @date_of_test_results as conceptId,
+                    r.date_of_result as valueDatetime, 
+                    1 as creator, 
+                    r.source_encounter_date as dateCreated,
```
Comment (mogoodrich) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy:102]: I would use result_entry_date here

File: etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                    r.source_encounter_date as obsDatetime, 
+                    1 as locationId, 
+                    @pcr_test_results as conceptId,
+                    case r.result 
+                        when 'negative' then  concept_from_mapping('CIEL', '664') 
+                        when 'positive'  then  concept_from_mapping('CIEL', '703') 
+                        when 'indeterminate'  then  concept_from_mapping('CIEL', '1138') 
+                    end as valueCoded, 
+                    1 as creator, 
+                    r.source_encounter_date as dateCreated,
```
Comment (mogoodrich) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy:134]: I would use result_entry_date here

File: etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            select  INFANT_ID, 
+                    SAMPLE_ID, 
+                    case 
+                        when (SAMPLE_DATE is not null ) then to_char(SAMPLE_DATE, 'yyyy-mm-dd') 
+                        else to_char(RESULT_ENTRY_DATE, 'yyyy-mm-dd')  
+                    end as encounter_date,         
+                    lower(RESULT) as result,        
+                    case 
+                        when (DATE_RESULT_RECEIVED is not null ) then to_char(DATE_RESULT_RECEIVED, 'yyyy-mm-dd') 
+                        else to_char(RESULT_ENTRY_DATE, 'yyyy-mm-dd')
```
Comment (mogoodrich) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy:42]: I think we can just use date_result_received here, and just keep null if there's no date result received (and then don't create an "date of results" obs below for these entries)...

File: etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                    1,
+                    1    
+            from hivmigration_pcr_tests r, hivmigration_infants i 
+            where r.source_infant_id = i.source_infant_id;                                                                                      
+        ''')
+
+        executeMysql("Create PCR results obs",'''
+
+            -- Add Date of Test Results
+            SET @date_of_test_results = (concept_from_mapping('PIH', 'DATE OF LABORATORY TEST'));
```
Comment (mogoodrich) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy:82]: Per my comment above, fine to just not create this obs for any results that don't have a "date of result received"

**Review by mogoodrich — CHANGES_REQUESTED** [review]
Generally LGTM @cioan , though I requested a few changes, thanks!

**Comment by cioan** [comment]
Thanks @mogoodrich ! I have addressed all your requests. Thanks!

**Review by mogoodrich — APPROVED** [review]
LGTM, thanks @cioan !

**Comment by cioan** [comment]
Thanks @mogoodrich !

**Pull request merged**